org.eclipse.jface.text/src/org/eclipse/jface/text/source/inlined/InlinedAnnotationSupport.java

```diff
@@ -333,9 +337,10 @@ public void mouseUp(MouseEvent e) {
 	private AnnotationPainter fPainter;
 
 	/**
-	 * Holds the current inlined annotations.
+	 * Holds the current inlined annotations based on their {@link Position}s in a HashMap to make
+	 * lookup, insertion and deletion in average-case time complexity of O(1).
 	 */
-	private Set<AbstractInlinedAnnotation> fInlinedAnnotations;
+	private Map<Position, AbstractInlinedAnnotation> fInlinedAnnotationByPositionMap;
```

Comment:
There can be multiple annotation on the same Position, so it should be `Map<Position, Collection<AbstractInlinedAnnotation>>`

Reply:
This map cannot work reliably.
Position is mutable on offset and length and the information is part of the hashCode. I suspect the annotations cannot be found anymore after some editing in the document happened. 

@mickaelistria please consider reverting this

Reply:
Thanks. I've checked in more details and the position that is part of the AbtractInlinedAnnotation seems to reference the same object than the one in the IAnnotationModel, which is updated on change. So it's not good to use it as Map key.
That kind of adds credit to my previous comment about storing positions in the IAnnotation being a bad idea in general and that instead this code should rely more on the IAnnotationModel when it comes to finding annotations or their position.

Reply:
> Would you like to push the revert PR 

I don't that'd add value. I cannot seem to find the revert button here in the MR immediately and a local setup just to issue a git revert && git push + making a MR... no thanks.
Please just go ahead with the revert.